src/audio/kpb.c

```diff
@@ -350,7 +356,11 @@ static void kpb_buffer_data(struct comp_data *kpb, struct comp_buffer *source)
 		space_avail = (int)hb->end_addr - (int)hb->w_ptr;
 
 		if (size_to_copy > space_avail) {
-			memcpy(hb->w_ptr, source->r_ptr, space_avail);
+			if (memcpy_s(hb->w_ptr, space_avail,
+			    source->r_ptr, space_avail)) {
+				trace_kpb_error("kpb_buffer_data()"
+						"unable to copy data");
```

Comment:
I guess we need some `return` or other *break* here in case of error

Reply:
I did not want to change KPB internal logic since i would need to change more functions upwards. @mrajwa What do You think we should do here ?

Reply:
After offline conversation with @mrajwa  will drop KPB changes since KPB will be heavily refactored.